sematic/db/queries.py

```diff
@@ -178,22 +178,38 @@ def save_resolution(resolution: Resolution) -> Resolution:
 
 def save_graph(runs: List[Run], artifacts: List[Artifact], edges: List[Edge]):
     """
-    Update a graph
+    Update a graph.
     """
     _assert_external_jobs_not_removed(runs)
     with db().get_session() as session:
         for run in runs:
             session.merge(run)
 
         for artifact in artifacts:
-            session.merge(artifact)
+            _save_artifact(artifact=artifact, session=session)
 
         for edge in edges:
             session.merge(edge)
 
         session.commit()
 
 
+def _save_artifact(artifact: Artifact, session: sqlalchemy.orm.Session) -> Artifact:
+    """
+    Saves or updates an Artifact, returning the actual persisted Artifact element.
+    """
+    previous_artifact = session.get(entity=Artifact, ident=artifact.id)
+
+    if previous_artifact is not None:
+        # we use content-addressed values for artifacts, with the id being
+        # generated from the type and value themselves
+        # there are currently no other fields that can be updated
+        logger.debug("Updating existing artifact %s", artifact.id)
+        artifact.created_at = previous_artifact.created_at
+
+    return session.merge(artifact)
```

Comment:
If the artifact already exists, the only thing we are changing is `updated_at` correct? Can we do that instead of settings `created_at` on the new artifact?